src/Microsoft.AspNet.Mvc.TagHelpers/LinkTagHelper.cs

```diff
@@ -229,15 +229,19 @@ public override void Process(TagHelperContext context, TagHelperOutput output)
 
             var builder = new DefaultTagHelperContent();
 
-            if (mode == Mode.Fallback && string.IsNullOrEmpty(HrefInclude) || mode == Mode.AppendVersion)
+            if (!string.IsNullOrEmpty(Href))
             {
-                // No globbing to do, just build a <link /> tag to match the original one in the source file.
-                // Or just add file version to the link tag.
-                BuildLinkTag(attributes, builder);
+                output.Attributes[HrefAttributeName].Value = AppendFileVersionIfApplicable(Href);
             }
-            else
+
+            if (mode == Mode.GlobbedHref || mode == Mode.Fallback && !string.IsNullOrEmpty(HrefInclude))
```

Comment:
FYI it would have been easier to understand your changes if you had left the condition in the original order. No change needed now but it was harder than necessary to see you'd simply remove the positive part of the earlier `if` / `else`.